text/0000-separate-context-propagation.md

```diff
@@ -0,0 +1,195 @@
+
+**Optional: Automated Context Management**  
+When possible, context should automatically be associated with program execution . Note that some languages do not provide any facility for setting and getting a current context. In these cases, the user is responsible for managing the current context. 
+
+**Optional: SetCurrentContext( context)**  
+To associate a context with program execution, the Context API provides a function which takes a Context.
+
+**Optional: GetCurrentContext() context**  
+To access the context associated with program execution, the Context API provides a function which takes no arguments and returns a Context.
+
+
+## Propagation API
+
+Distributed applications send data to downstream processes via propagators, functions which read and write application context into RPC requests. Each distributed application creates a set of propagators for every type of supported medium - currently HTTP and Binary.
+
+**Inject( context, request)**  
+To send the data for all distributed applications downstream to the next process, the Propagation API provides a function which takes a context and a request, and mutates the request to include the encoded context. The canonical representation of a request is as a map.
+
+**Extract( context, request) context**  
+To receive data injected by prior upstream processes, the Propagation API provides a function which takes a context and a request, and returns an updated context.
+
+**RegisterPropagator( type, inject, extract)**  
+In order for the application layer to function correctly, Propagation choices must be syncronized between all processes in the distributed system, and multiple applications must be able to inject and extract their context into the same request. To meet these requirements, the Propagation API provides a function which registers a set of propagators, which will all be executed in order when the future calls to inject and extract are made. A canonical propagator consists of an inject and an extract function.
+
+OpenTelemetry currently contains two types of Propagators:
+
+* **HTTP** - context is written into and read from a map of HTTP headers.
+* **Binary** - context is serialized into and deserialized from a stream of bytes.
+
+# Internal details
+
+![drawing](img/context_propagation_details.png)
+
+## Context details
+OpenTelemetry currently implements three context types of context propagation.
+
+**Span Context -** Observability data used by the tracing system. The readable attributes are defined to match those found in the  W3C **traceparent** header. Span Context is used as labels for metrics and traces. This can quickly add overhead when propagated in-band. But, because this data is write-only, how this information is transmitted remains undefined. 
+
+**Correlation Context -** Transaction-level observability data, which can be applied as labels to spans and metrics. 
```

Comment:
can we better explain the difference between correlation context and span context?

Reply:
Yeah totally. Please let me know if the new descriptions are better.

Reply:
Much better.